code/modules/explorer_drone/adventure.dm

```diff
@@ -0,0 +1,422 @@
+#define NODE_ON_EXIT_EFFECTS_FIELD "on_exit_effects"
+
+#define CHOICE_KEY_FIELD "key"
+#define CHOICE_NAME_FIELD "name"
+#define CHOICE_ON_SELECTION_EFFECT_FIELD "on_selection_effects"
+#define CHOICE_REQUIREMENTS_FIELD "requirements"
+#define CHOICE_EXIT_NODE_FIELD "exit_node"
+#define CHOICE_DELAY_FIELD "delay"
+#define CHOICE_DELAY_MESSAGE_FIELD "delay_message"
+
+#define EFFECT_TYPE_FIELD "effect_type"
+#define EFFECT_QUALITY_FIELD "quality"
+#define EFFECT_VALUE_FIELD "value"
+#define EFFECT_VALUE_VALUE_TYPE_FIELD "value_type"
+#define TRIGGER_NAME_FIELD "name"
+#define TRIGGER_REQUIREMENTS_FIELD "requirements"
+#define TRIGGER_ON_TRIGGER_EFFECTS_FIELD "on_trigger_effects"
+#define TRIGGER_TARGET_NODE_FIELD "target_node"
+
+#define REQ_GROUP_REQUIREMENTS_FIELD "requirements"
+#define REQ_GROUP_GROUP_TYPE_FIELD "group_type"
+
+#define REQ_QUALITY_FIELD "quality"
+#define REQ_VALUE_FIELD "value"
+#define REQ_OPERATOR_FIELD "operator"
+
+GLOBAL_LIST_EMPTY(explorer_drone_adventures)
+
+
+/proc/load_adventures()
+	. = list()
+	for(var/filename in flist(ADVENTURE_DIR))
+		var/datum/adventure/A = try_loading_adventure(filename)
+		if(A)
+			. += A
+
+/proc/try_loading_adventure(filename)
+	var/list/json_data = json_load(ADVENTURE_DIR+filename)
+	if(!islist(json_data))
+		CRASH("Invalid JSON in adventure file [replacetext(filename,ADVENTURE_DIR,"")]")
+	//Basic validation of required fields, don't even bother loading if they are missing.
+	var/static/list/required_fields = list(ADVENTURE_NAME_FIELD,ADVENTURE_STARTING_NODE_FIELD,ADVENTURE_NODES_FIELD)
+	for(var/field in required_fields)
+		if(!json_data[field])
+			CRASH("Adventure file [replacetext(filename,ADVENTURE_DIR,"")] missing [field] value")
+
+	var/datum/adventure/loaded_adventure = new
+	//load properties
+	loaded_adventure.starting_node = json_data[ADVENTURE_STARTING_NODE_FIELD]
+	loaded_adventure.name = json_data[ADVENTURE_NAME_FIELD]
+	loaded_adventure.required_site_traits = json_data[ADVENTURE_REQUIRED_SITE_TRAITS_FIELD]
+	loaded_adventure.band_modifiers = json_data[ADVENTURE_SCAN_BAND_MODS_FIELD]
+	loaded_adventure.loot_categories = json_data[ADVENTURE_LOOT_FIELD]
+	loaded_adventure.starting_qualities = json_data[ADVENTURE_STARTING_QUALITIES_FIELD]
+	loaded_adventure.deep_scan_description = json_data[ADVENTURE_DEEP_SCAN_DESCRIPTION]
+
+	for(var/list/node_data in json_data[ADVENTURE_NODES_FIELD])
+		var/datum/adventure_node/node = try_loading_node(node_data)
+		if(node)
+			if(loaded_adventure.nodes[node.id])
+				CRASH("Duplicate [node.id] node in [replacetext(filename,ADVENTURE_DIR,"")] adventure")
+			loaded_adventure.nodes[node.id] = node
+	loaded_adventure.triggers = json_data[ADVENTURE_TRIGGERS_FIELD]
+	if(!loaded_adventure.validate())
+		CRASH("Validation failed for [replacetext(filename,ADVENTURE_DIR,"")] adventure")
+	return loaded_adventure
+
+/proc/try_loading_node(node_data)
+	if(!islist(node_data))
+		CRASH("Invalid adventure node data")
```

Comment:
I think these crashes should likely provide better traces of where the invalid data actually is, rather than just something vague like this.